src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContextSerializer.cs

```diff
@@ -25,25 +24,97 @@ private DesigntimeLicenseContextSerializer()
         /// </summary>
         public static void Serialize(Stream o, string cryptoKey, DesigntimeLicenseContext context)
         {
-            IFormatter formatter = new BinaryFormatter();
+            if (LocalAppContextSwitches.BinaryFormatterEnabled)
+            {
+                IFormatter formatter = new BinaryFormatter();
 #pragma warning disable SYSLIB0011 // Issue https://github.com/dotnet/runtime/issues/39293 tracks finding an alternative to BinaryFormatter
-            formatter.Serialize(o, new object[] { cryptoKey, context._savedLicenseKeys });
+                formatter.Serialize(o, new object[] { cryptoKey, context._savedLicenseKeys });
 #pragma warning restore SYSLIB0011
+            }
+            else
+            {
+                using (BinaryWriter writer = new BinaryWriter(o, encoding: Text.Encoding.UTF8, leaveOpen: true))
+                {
+                    writer.Write(-1); // flag to identify BinaryWriter
```

Comment:
This is a breaking change. However, AFAICT, nothing in the libs calls this method. Customers who currently try to deserialize a stream that was serialized without this flag will need to call this method again. This sounds reasonable to me. 

If we want to avoid this breaking change, I could look into whether we can examine the first 17 bytes of a stream to determine if it was formatted with Binary Formatter. I'm curious to hear about @GrabYourPitchforks's opinion here. IIRC, nobody is using this method currently, so we might be able to get away with this change.